components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -11,6 +11,18 @@ pub fn length(arg: &Option<Bytes>) -> Result<Option<i64>> {
     Ok(arg.as_ref().map(|bytes| bytes.len() as i64))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn ascii(arg: &Option<Bytes>) -> Result<Option<i64>> {
+    Ok(arg.as_ref().map(|bytes| {
```

Comment:
Prefer to:
```rust
Ok(arg.as_ref().get(0).map(|b|i64::from(*b)))
```

Reply:
thx for help. But I found that this way of writing it, return type will change to `Result<Option<Option<i64>>>`, it should use `arg.as_ref().and_then()`. and in case when `Bytes` is empty, expect return` Ok<Some(0)>`, but found `Ok<None>`

Reply:
OK, I have approve this PR.b(let's refine it in the future via `flatten` (https://github.com/rust-lang/rust/pull/64747).)